source/extensions/filters/http/aws_lambda/aws_lambda_filter.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/http/filter.h"
+
+#include "extensions/common/aws/signer.h"
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/strings/str_split.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AwsLambdaFilter {
+
+class Arn {
+public:
+  Arn(absl::string_view partition, absl::string_view service, absl::string_view region,
+      absl::string_view account_id, absl::string_view resource_type,
+      absl::string_view function_name)
+      : partition_(partition), service_(service), region_(region), account_id_(account_id),
+        resource_type_(resource_type), function_name_(function_name) {}
+
+  const std::string& partition() const { return partition_; }
+  const std::string& service() const { return service_; }
+  const std::string& region() const { return region_; }
+  const std::string& accountId() const { return account_id_; }
+  const std::string& resourceType() const { return resource_type_; }
+  const std::string& functionName() const { return function_name_; }
+
+private:
+  std::string partition_;
+  std::string service_;
+  std::string region_;
+  std::string account_id_;
+  std::string resource_type_;
+  std::string function_name_; // resource_id
+};
+
+/**
+ * Parses the input string into a structured ARN.
+ *
+ * The format is expected to be as such:
+ * arn:partition:service:region:account-id:resource-type:resource-id
+ *
+ * Lambda ARN Example:
+ * arn:aws:lambda:us-west-2:987654321:function:hello_envoy
+ */
+absl::optional<Arn> parse_arn(absl::string_view arn);
```

Comment:
Why not make this a class method of Arn ?

Reply:
Because it doesn't need access to Arn's private members.

Reply:
Well I meant like a static factory function. But cool either way